### ISSUE TITLE
What is different between point-collocation and spectral projection method? 

### ISSUE THREAD

**Issue by nguyenhoaiThanhbk2811** [jonathf/chaospy#103]
Hi you.
I am doing a comparison between two method point collocation and spectral project method. What is different between the two methods? and which the method has the best result? when do we use them?
I have read the tutorial of yours.  your conclusion is the spectral Project have the result more than point collocation. but when I try with my research then I have the opposite conclusion with you. have you the paper about this issue?

I have been used the tool NISP on the Scilab software to implement the spectral Projection method to compare with the point collocation method in your tool.

"https://chaospy.readthedocs.io/en/master/tutorial.html"

I am also reading the paper may be the paper using your tool but it is not compared between the two methods.
"https://www.biorxiv.org/content/biorxiv/early/2018/03/05/274779.full.pdf"
thank you.
HoaiThanh

**Comment by jonathf** [comment]
First of all, if you used NISP, you haven't used spectral projection you have ben using psuedo-spectral projection (as I call it) or Non-intrusive spectral projection (as they call it). Actual spectral projection, often called intrusive Galerkin method.

The difference between the two methods it can be summarized as follows:
* Collocation method estimate the Fourier coefficients through least squares regression of evaluations of a model.
* Psuedo-spectral method estimates the same coefficients through nummerical integration by noting that each coefficient have the form: `<u,P>/<P,P>` and applying quadrature rules.

As a rule of thumb, pseudo-spectral method has the potential for higher accuracy, but is also more fragile. You are more likely to get stable good results with collocation. This is especially true for non-linear problems.

I am assuming that you are doing this for research, and the fact that you are asking this question, you are looking to get an overview. If that is correct, I can highly recommend the book "Numerical Methods for Stochastic Computations -- A Spectral Method Aproach" by Dongbin Xiu. It explains the concepts a lot better and in details.

**Comment by nguyenhoaiThanhbk2811** [comment]
your mean is the NISP and Pseudo-spectral projection methods are two different methods. If it is the difference, you could give me your recommendations when do we use the Pseudo-spectral projection. 

But I have been reading the paper they said that " The non-intrusive methods for doing this can be divided into two classes, point-collocation methods, and pseudo-spectral projection methods, both of which are implemented in Uncertainty."

+ Point collocation is the default method used in Uncertainty. This method is based on demanding that the polynomial approximation is equal to the model output evaluated at a set of collocation nodes drawn from the joint probability density function ρQ
+Pseudo-spectral projection methods are based on the least squares minimization in the orthogonal polynomial space and calculate the expansion coefficients Cn through numerical integration.

---> Of these two methods, point collocation is robust toward invalid model evaluations as long as the number of remaining evaluations is high enough, while spectral projection is not. ( could you explain for me this is the sentence)
thank you very much.
hoai thanh

**Comment by jonathf** [comment]
No, I am saying that spectral projection is an ambiguous term and you shouldn't use the term without "pseudo-" or "non-intrusive" prefix is in order to not mistake it from the Galerkin methods.

Sounds about right.

I would like to note that I think we have gotten to the point that it is worth saying using an issue tracker of a software you are using might not be the correct place for such a conversation. I am here to help you if you need help implementing a specific method using Chaospy. I am not here to help you understand the underlying theory. That is why I recommended the book.
